File: Cliente-WinForms/model/ResumenCuentaDto.cs
using Newtonsoft.Json;

namespace WayBankClient.model
{
    /// <summary>
    /// DTO — Consulta personalizada #2 (TERCER PROTOTIPO).
    ///
    /// "Una consulta debe mostrar datos de la tabla maestro y dos de detalle."
    ///
    /// Devuelve datos de CUENTA_AHORROS (maestro) más dos campos agregados
    /// calculados sobre MOVIMIENTO (detalle):
    ///   • totalMovimientos — COUNT de movimientos asociados
    ///   • totalCreditos    — SUM de montos CREDITO
    ///
    /// Endpoint: GET /cuentas/{numero}/resumen en MS-CuentaAhorros (8080).
    /// </summary>
    public class ResumenCuentaDto
    {
        // ---- Datos del MAESTRO (CUENTA_AHORROS) ----
        [JsonProperty("numeroCuenta")]
        public int NumeroCuenta { get; set; }

        [JsonProperty("titular")]
        public string Titular { get; set; }

        [JsonProperty("saldo")]
        public double Saldo { get; set; }

        [JsonProperty("estado")]
        public string Estado { get; set; }

        [JsonProperty("tasaInteres")]
        public double TasaInteres { get; set; }

        // ---- Datos del DETALLE (MOVIMIENTO) — agregados ----

        /// <summary>COUNT de movimientos asociados a esta cuenta.</summary>
        [JsonProperty("totalMovimientos")]
        public long TotalMovimientos { get; set; }

        /// <summary>SUM de montos con tipo CREDITO.</summary>
        [JsonProperty("totalCreditos")]
        public double TotalCreditos { get; set; }
    }
}

File: MS-CuentaAhorros/src/main/java/com/jacs/proyectomicroservicios/dto/ResumenCuentaDTO.java
package com.jacs.proyectomicroservicios.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO — Consulta personalizada 2 (TERCER PROTOTIPO).
 *
 * "Una de las consultas debe permitir mostrar los datos de la tabla maestro
 *  y dos de detalle."
 *
 * Retorna los campos principales de CUENTA_AHORROS (maestro) más dos
 * campos agregados de MOVIMIENTO (detalle):
 *   - totalMovimientos : cantidad de movimientos registrados
 *   - totalCreditos    : suma de montos con tipo CREDITO
 *
 * Endpoint: GET /cuentas/{numero}/resumen
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ResumenCuentaDTO {

    // ---- Datos de la tabla maestro (CUENTA_AHORROS) ----

    private int    numeroCuenta;
    private String titular;
    private double saldo;
    private String estado;
    private double tasaInteres;

    // ---- Dos datos de la tabla detalle (MOVIMIENTO) ----

    /** Cantidad total de movimientos asociados a la cuenta. */
    private long totalMovimientos;

    /** Suma de montos de todos los movimientos tipo CREDITO. */
    private double totalCreditos;
}
